stdlib/src/lib.rs

```diff
@@ -86,7 +86,14 @@ pub unsafe extern "C" fn __quantum__rt__fail(str: *const CString) {
 #[cfg(feature = "message-support")]
 #[no_mangle]
 pub unsafe extern "C" fn __quantum__rt__message(str: *const CString) {
-    println!("{}", (*str).to_str().expect("Unable to convert string"));
+    crate::output_recording::record_output_str(&format!(
```

Comment:
nitpick: it's nice to have these be `use` based rather than fully qualified so that later we can look at the top of the file and see all the dependencies.